sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ClientEncryptionPolicy.java

```diff
@@ -0,0 +1,121 @@
+        this.validateIncludedPaths(paths);
+        this.includedPaths = paths;
+        this.policyFormatVersion = 1;
+    }
+
+    /**
+     * Constructor.
+     */
+    public ClientEncryptionPolicy() {
+        this.jsonSerializable = new JsonSerializable();
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param jsonString the json string that represents the client encryption policy.
+     */
+    ClientEncryptionPolicy(String jsonString) {
+        this.jsonSerializable = new JsonSerializable(jsonString);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param objectNode the object node that represents the client encryption policy.
+     */
+    ClientEncryptionPolicy(ObjectNode objectNode) {
+        this.jsonSerializable = new JsonSerializable(objectNode);
+    }
+
+    public List<ClientEncryptionIncludedPath> getIncludedPaths() {
+        return this.includedPaths;
+    }
+
+    int getPolicyFormatVersion() {
+        return policyFormatVersion;
+    }
+
+    void setPolicyFormatVersion(int policyFormatVersion) {
+        this.policyFormatVersion = policyFormatVersion;
+    }
+
+    private void validateIncludedPaths(List<ClientEncryptionIncludedPath> clientEncryptionIncludedPath) {
+        List<String> includedPathsList = new ArrayList<>();
+        for (ClientEncryptionIncludedPath path : clientEncryptionIncludedPath) {
+            this.validateClientEncryptionIncludedPath(path);
+            if (includedPathsList.contains(path.path)) {
+                throw new IllegalArgumentException("Duplicate Path found in clientEncryptionIncludedPath.");
+            }
+
+            includedPathsList.add(path.path);
+        }
+    }
+
+    private void validateClientEncryptionIncludedPath(ClientEncryptionIncludedPath clientEncryptionIncludedPath) {
+        if (clientEncryptionIncludedPath == null) {
+            throw new IllegalArgumentException("clientEncryptionIncludedPath is null");
+        }
+
+        if (StringUtils.isEmpty(clientEncryptionIncludedPath.path)) {
+            throw new IllegalArgumentException("path in clientEncryptionIncludedPath is empty");
+        }
+
+        if (clientEncryptionIncludedPath.path.charAt(0) != '/'
+            || clientEncryptionIncludedPath.path.lastIndexOf('/') != 0
+            || clientEncryptionIncludedPath.path.substring(1).equals("id")) {
+            throw new IllegalArgumentException("Invalid path " + clientEncryptionIncludedPath.path);
+        }
+
+        if (StringUtils.isEmpty(clientEncryptionIncludedPath.clientEncryptionKeyId)) {
+            throw new IllegalArgumentException("clientEncryptionKeyId in clientEncryptionIncludedPath is empty");
+        }
+
+        if (StringUtils.isEmpty(clientEncryptionIncludedPath.encryptionType)) {
+            throw new IllegalArgumentException("encryptionType in clientEncryptionIncludedPath is empty");
+        }
+
+        if (!clientEncryptionIncludedPath.encryptionType.equals("Deterministic") &&
```

Comment:
can you use defined constant for the magic string: "Deterministic" and elsewhere

Reply:
Done